src/strategies/GammaQuickSwapMerklFarmStrategy.sol

```diff
@@ -291,35 +292,78 @@ contract GammaQuickSwapMerklFarmStrategy is LPStrategyBase, MerklStrategyBase, F
         override(StrategyBase, LPStrategyBase)
         returns (uint[] memory amountsConsumed, uint value)
     {
-        // alternative calculation: beefy-contracts/contracts/BIFI/strategies/Gamma/StrategyQuickGamma.sol
         GammaQuickSwapFarmStrategyStorage storage $ = _getGammaQuickStorage();
         StrategyBaseStorage storage _$ = _getStrategyBaseStorage();
         amountsConsumed = new uint[](2);
         address[] memory _assets = assets();
         address underlying_ = _$._underlying;
+
         (uint amount1Start, uint amount1End) = $.uniProxy.getDepositAmount(underlying_, _assets[0], amountsMax[0]);
+        IFactory.Farm memory farm = _getFarm();
+
+        farm.nums[0] == ALMPositionNameLib.STABLE
+            ? _handleStableAmounts(amountsMax, $, underlying_, _assets, amount1Start, amount1End, amountsConsumed)
+            : _handleNonStableAmounts(amountsMax, $, underlying_, _assets, amount1Start, amount1End, amountsConsumed);
+
+        // calculate shares
+        value = _calculateShares(amountsConsumed, underlying_);
+    }
+
+    function _handleStableAmounts(
+        uint[] memory amountsMax,
+        GammaQuickSwapFarmStrategyStorage storage $,
+        address underlying_,
+        address[] memory _assets,
+        uint amount1Start,
+        uint amount1End,
+        uint[] memory amountsConsumed
+    ) internal view {
+        amountsConsumed[1] = amountsMax[1];
+        amountsConsumed[0] = amountsMax[0];
+        (, uint amount0End) = $.uniProxy.getDepositAmount(underlying_, _assets[1], amountsMax[1]);
+        // Inline the assignment and condition with a ternary operator
+        amountsConsumed[1] = (amountsMax[1] > amount1End) ? amount1End : amountsMax[1];
+
+        // Check the second condition within another ternary operation
+        (amountsMax[1] <= amount1Start) ? amountsConsumed[0] = amount0End : amountsConsumed[0];
+
+        // Set amountsConsumed[1] to amountsMax[1] only if the second condition holds true
+        amountsConsumed[1] = (amountsMax[1] <= amount1Start) ? amountsMax[1] : amountsConsumed[1];
+
+        // Ensure amountsConsumed[0] does not exceed amount0End
+        amountsConsumed[0] = (amountsConsumed[0] > amount0End) ? amount0End : amountsConsumed[0];
+    }
+
+    function _handleNonStableAmounts(
+        uint[] memory amountsMax,
+        GammaQuickSwapFarmStrategyStorage storage $,
+        address underlying_,
+        address[] memory _assets,
```

Comment:
see slither alert about arg name convention

Reply:
fixed that problem.